File: src/main/java/org/code4everything/hutool/converter/ObjectPropertyConverter.java
package org.code4everything.hutool.converter;

import cn.hutool.core.bean.BeanUtil;
import org.code4everything.hutool.Converter;

/**
 * @author pantao
 * @since 2020/11/1
 */
public class ObjectPropertyConverter implements Converter<Object> {

    @Override
    public Object string2Object(String string) {
        throw new UnsupportedOperationException("not implemented");
    }

    @Override
    public String object2String(Object object) {
        MapConverter mapConverter = new MapConverter();
        return mapConverter.object2String(BeanUtil.beanToMap(object));
    }
}

File: src/main/kotlin/org/code4everything/hutool/converter/ObjectPropertyConverter.kt
package org.code4everything.hutool.converter

import cn.hutool.core.bean.BeanUtil
import org.code4everything.hutool.Converter

class ObjectPropertyConverter : Converter<Any> {

    override fun string2Object(string: String): Any = TODO("not implemented")

    override fun object2String(any: Any): String = MapConverter().object2String(BeanUtil.beanToMap(any))
}
